pages/guide/07-full-stack-deployments.md

```diff
@@ -1,4 +1,156 @@
-> **Note**: this guide is a work in progress and is not currently complete. Content
-> here will change and be added over time. In the meantime, you can check out the
-> [Gleam Discord server](https://discord.gg/Fm8Pwmy) if you have any questions about
-> how to deploy full stack applications with Lustre.
+# 07 Full-Stack Deployments
+
+If you have built a full-stack Lustre application then you can use Docker and a platform provider such as [Fly.io](https://fly.io/) to deploy your application.
+
+In this guide, you will learn how to manually deploy a Lustre full-stack application to Fly.io using the `flyctl` CLI. You will also learn how to leverage GitHub Actions for continuous deployments.
+
+If you are planning on deploying a Lustre SPA without a backend, you probably don't want to follow this guide! Instead, refer to the [Lustre SPA deployment guide](./04-spa-deployments.html).
+
+## Prerequisites
+
+In this guide, we assume that you have a Lustre full-stack application with the following _monorepo_ structure consisting of three separate Gleam projects:
+
+```sh
+common/
+client/
+server/
+```
+
+Here, the `common/` directory contains the Lustre application logic which is agnostic to the platform it runs on.
+
+The `client/` directory contains the client-side code, which is typically only responsible for the boilerplate necessary to run the Lustre application in the browser. The client-side code uses the JavaScript target and is built with `lustre_dev_tools` with the `--outdir` flag set to `../server/priv/static`.
+
+The `server/` directory contains the server-side code and runs the same Lustre application as a server component. The server-side code uses the Erlang target.
+
+In this guide, we assume this monorepo structure because it happens to be the same structure used in the [Lustre full-stack application guide](./06-full-stack-applications.html).
```

Comment:
I did not go into much detail on how to structure the full stack application here because:

a) I am not an expert on Lustre full stack applications
b) I felt like that will be covered by what I linked to here in the future

but whether what was a wise decision or not is of course up for debate